### PR TITLE
Conversation list

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Based on #2291

### CODE REVIEW

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+			<ActionButton v-if="canFavorite"
+				:icon="iconFavorite"
+				@click.prevent.exact="toggleFavoriteConversation">
+				{{ labelFavorite }}
+			</ActionButton>
+
+			<!-- FIXME Should be a real separator -->
+			<ActionText>
+				FIXME I'm a separator
+			</ActionText>
```
Comment (nickvergessen) [inline src/components/Navigation/ConversationsList/ConversationsListItem.vue]: Separator

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+			<ActionButton
+				icon="icon-sound-off"
+				@click.prevent.exact="setNotificationLevel(3)">
+				{{ t('spreed', 'Off') }}
+			</ActionButton>
+
+			<!-- FIXME Should be a real separator -->
+			<ActionText>
+				FIXME I'm a seperator
+			</ActionText>
```
Comment (nickvergessen) [inline src/components/Navigation/ConversationsList/ConversationsListItem.vue]: Separator

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+			<ActionButton
+				icon="icon-user"
+				@click.prevent.exact="setNotificationLevel(2)">
+				{{ t('spreed', '@-mentions only') }}
+			</ActionButton>
+			<ActionButton
+				icon="icon-sound-off"
+				@click.prevent.exact="setNotificationLevel(3)">
+				{{ t('spreed', 'Off') }}
+			</ActionButton>
```
Comment (nickvergessen) [inline src/components/Navigation/ConversationsList/ConversationsListItem.vue]: Can not set the active class, ActionRadio however makes us loose the icon

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+			:highlighted="true">
+			{{ item.unreadMessages }}
+		</AppNavigationCounter>
+		<template slot="actions">
+			<ActionButton v-if="canFavorite"
+				:icon="iconFavorite"
+				@click.prevent.exact="toggleFavoriteConversation">
+				{{ labelFavorite }}
+			</ActionButton>
+
```
Comment (nickvergessen) [inline src/components/Navigation/ConversationsList/ConversationsListItem.vue]: ClipBoard action is missing

File: src/components/Navigation/ConversationsList/ConversationsList.vue [hunk 5]
```diff
@@ -21,57 +21,22 @@
...
-import { joinConversation, removeCurrentUserFromConversation } from '../../../services/participantsService'
 
 export default {
 	name: 'ConversationsList',
 	components: {
-		Avatar,
-		AppNavigationCounter,
-		ActionButton,
-		AppContentListItem
+		ConversationsListItem
```
Comment (ma12-co) [inline src/components/Navigation/ConversationsList/ConversationsList.vue]: Actually maybe just `Conversation` is eough here

File: src/components/ConversationsIcon.vue [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+		:size="44"
+		:user="item.displayName"
+		:display-name="item.displayName" />
+</template>
+
+<script>
+import Avatar from 'nextcloud-vue/dist/Components/Avatar'
+
+export default {
+	name: 'ConversationsIcon',
```
Comment (ma12-co) [inline src/components/ConversationsIcon.vue]: `ConversationIcon`?

Comment (nickvergessen) [reply src/components/ConversationsIcon.vue]: Well, that should be fixed in #2291  then :P

Comment (ma12-co) [reply src/components/ConversationsIcon.vue]: oh yes!

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue (hunk 1 above)
Comment (nickvergessen) [reply src/components/Navigation/ConversationsList/ConversationsListItem.vue]: Old template code
https://github.com/nextcloud/spreed/blob/e94805d/js/views/templates/roomlistview.handlebars#L46

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+			<!-- FIXME Should be a real separator -->
+			<ActionText>
+				FIXME I'm a separator
+			</ActionText>
+
+			<ActionText>
+				{{ t('spreed', 'Chat notifications') }}
+			</ActionText>
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply src/components/Navigation/ConversationsList/ConversationsListItem.vue]: Old template code
https://github.com/nextcloud/spreed/blob/e94805d/js/views/templates/roomlistview.handlebars#L47

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue (hunk 3 above)
Comment (nickvergessen) [reply src/components/Navigation/ConversationsList/ConversationsListItem.vue]: Old template code
https://github.com/nextcloud/spreed/blob/e94805d/js/views/templates/roomlistview.handlebars#L48-L65

Comment (nickvergessen) [reply src/components/Navigation/ConversationsList/ConversationsListItem.vue]: @ma12-co I'm very unsure about this one. And idea what would be the best way to allow this?

**Comment by nickvergessen** [comment]
Btw at least for me it takes quite a long time, until the icons in the actions menu are shown. Why is that? they come from a SCSS file and are ordinary SVGs. They should be loaded once and then cached?

File: src/components/Navigation/ConversationsList/ConversationsListItem.vue (hunk 3 above)
Comment (ma12-co) [reply src/components/Navigation/ConversationsList/ConversationsListItem.vue]: @nickvergessen  we've been discussing this yesterday at the design meeting and I'm going to add this capability to the actions component.

**Comment by ma12-co** [comment]
>  at least for me it takes quite a long time, until the icons in the actions menu are shown. 

not sure why, in my machine they show up instantly without any lag

**Comment by nickvergessen** [comment]
So let's merge this and do the rest in a follow up?

**Comment by ma12-co** [comment]
@nickvergessen Lemme test it again, can you re-request review?

**Pull request merged**